extensions/amp-instagram/1.0/test/test-instagram.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {Instagram} from '../instagram';
+import {mount} from 'enzyme';
+
+describes.sandboxed('Instagram preact component v1.0', {}, (env) => {
+  it('Normal render', () => {
+    const el = document.createElement('div');
+    document.body.appendChild(el);
+    const wrapper = mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        style={{
+          'width': 500,
+          'height': 600,
+        }}
+      />,
+      {attachTo: el}
+    );
+    expect(wrapper.props().shortcode).to.equal('B8QaZW4AQY_');
+    expect(wrapper.find('iframe').prop('src')).to.equal(
+      'https://www.instagram.com/p/B8QaZW4AQY_/embed/?cr=1&v=12'
+    );
+    expect(wrapper.find('iframe').prop('style').width).to.equal('100%');
+    expect(wrapper.find('iframe').prop('style').height).to.equal('100%');
+    expect(wrapper.find('div')).to.have.lengthOf(2);
+  });
+
+  it('Render with caption', () => {
+    const el = document.createElement('div');
+    document.body.appendChild(el);
+    const wrapper = mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        captioned
+        style={{'width': 500, 'height': 705}}
+      />,
+      {attachTo: el}
+    );
+    expect(wrapper.props().shortcode).to.equal('B8QaZW4AQY_');
+    expect(wrapper.find('iframe').prop('src')).to.equal(
+      'https://www.instagram.com/p/B8QaZW4AQY_/embed/captioned/?cr=1&v=12'
+    );
+    expect(wrapper.find('iframe').prop('style').width).to.equal('100%');
+    expect(wrapper.find('iframe').prop('style').height).to.equal('100%');
+    expect(wrapper.find('div')).to.have.lengthOf(2);
+  });
+
+  it('Resize prop is called', async () => {
+    const wrapper = mount(
+      <Instagram
+        shortcode="B8QaZW4AQY_"
+        style={{'width': 500, 'height': 705}}
+        requestResize={env.sandbox.spy()}
+      />,
+      {attachTo: document.body}
+    );
+
+    await new Promise((resolve) =>
```

Comment:
Do you actually impress a live iframe here? Or is it indirected somehow?

Reply:
This is a live iframe here. When the test is run without `--headless`, the iframe is visible on the test webpage.

Reply:
/cc @alanorozco, but it seems like running a live iframe for a unit test is going to cause problems. How was the 0.1 tested? We also have a bunch of other iframe-based components following this one, so it's best to figure it out now.

Reply:
Looking at the 0.1 tests, a live iframe is used for many of the tests. The iframe is created, appended to the body, and loaded in `getIns` here: https://github.com/ampproject/amphtml/blob/83d641b7b3057395615d6fbb37e91ba8e027ec39/extensions/amp-instagram/0.1/test/test-amp-instagram.js#L34

A benefit I see of using a live iframe is that it always reflects the current accuracy of the component in handling Instagram's messages. If Instagram changes the way it sends messages, the test failure will notify us immediately. 

However, that can also be a con, since relying on Instagram for tests to pass can be unreliable (e.g. Instagram is down). What do you think?

Reply:
That's good for an integration test. But I'm not certain that we are "allowed" to do so in unit tests somewhere on Travis. E.g. Instagram can blacklist those IPs if they feel like it's not a proper use. Such tests would also be very flaky. We need to confirm with @alanorozco and @ampproject/wg-infra what's the right path here.

Reply:
I actually don't think 0.1 tests the live iframe. Furthermore, I don't think that's even allowed. This code blocks all external iframes loading: https://github.com/ampproject/amphtml/blob/e3c5533c7d082fdac5cd88959c8304db378b14cb/testing/iframe.js#L513

So keep it in mind. Even if you think it's loading, it's not in fact.

Reply:
Unit tests almost certainly shouldn't be making any real web requests (in fact, echoing Dima above, I believe we block outgoing network requests in some testing environments). The right approach here is probably mocking the network response or, if there's no way to do so sufficiently, adding an endpoint to `build-system/server/app.js` (example can be found there of `/adzerk/*` simulating the Adzerk server). At this point it starts to reach the point of testing the server stub, so mocking the different expected responses within the test should be preferred.

For ensuring it works with live response data, I think pages under `examples/` may be run with network access in Visual Diff testing, but I'm not sure.

Reply:
Thank you all for the explanations! I learned something new to put in my design doc. I'll look into these approaches and see what is the best way to go moving forward.